dynamics/j2_dynamics.py

```diff
@@ -0,0 +1,102 @@
+"""
+J2 dynamics and jacobian
+"""
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+from brahe.constants import GM_EARTH, J2_EARTH, R_EARTH
+
+
+def j2_dynamics(r: jnp.ndarray) -> jnp.ndarray:
+    """
+    Provide the J2 pertubation dynamics
+
+    :param r: position vector
+    :return: J2 pertubation dynamics
+    """
+    r_norm = jnp.linalg.norm(r)
+
+    F = 3 * GM_EARTH * J2_EARTH * R_EARTH**2 / (2 * r_norm**5)
```

Comment:
I would suggest lumping all of these constants together into a single module-level constant, so that we don't keep recomputing it.